src/collections/signups/hooks/send-confirmation-email.ts

```diff
@@ -33,9 +35,23 @@ export const sendConfirmationEmail: CollectionAfterChangeHook<Signup> = async ({
       ? await payload.findByID({ collection: "users", id: doc.user })
       : doc.user;
 
-  if (!event) {
+  const eventSignupEmailPreference = await payload.find({
+    collection: "user-notification-preferences",
+    where: {
+      user: { equals: user.id },
+      type: { equals: notificationTypes.EVENT_SIGNUP },
+      channel: { equals: notificationChannels.EMAIL },
+    },
+  });
+
+  if (
+    !event ||
+    (eventSignupEmailPreference &&
```

Comment:
Is there a scenario where `eventSignupEmailPreference` is undefined? Looking at the docs it seems to me that `docs` will be an empty array if 0 docs match the query.

```suggestion
    (eventSignupEmailPreference.docs.length > 0 &&
```